src/shared/ui/pickerPrompter.ts

```diff
@@ -261,21 +279,23 @@ export class QuickPickPrompter<T> extends Prompter<T> {
     public loadItems(items: Promise<DataQuickPickItem<T>[]> | DataQuickPickItem<T>[]): Promise<void> {
         const picker = this.quickPick
 
+        if (this.hasPlaceholder) {
+            this.clearItems()
+        }
+
         if (items instanceof Promise) {
             picker.busy = true
             picker.enabled = false
+            this.pendingPromises += 1
 
             return items.then(items => {
-                const previousSelected = picker.activeItems
-                picker.items = picker.items.concat(items)
-                this.selectItems(...previousSelected)
-                picker.busy = false
+                this.pendingPromises -= 1
+                picker.busy = this.pendingPromises !== 0
+                this.appendItems(items)
                 picker.enabled = true
```

Comment:
This shouldn't be enabled unless `!picker.busy`

Reply:
Yup enabled should instead be an option for the caller to set.